### ISSUE TITLE
Have you resolved the unreasonable crf output sequence issue

### ISSUE THREAD

**Issue by betterbear** [createmomo/CRF-Layer-on-the-Top-of-BiLSTM#4]
Just wondering have you resolved the unreasonable crf output sequence such as:
word1 O
word2 I-PERSON
word3 I-ORG
If you have solved this issue, how?

**Comment by createmomo** [comment]
hi betterbear, in my case, the unreasonable output sequence was solved very well by the CRF layer. TO be detailed, the unreasonable output sequence is from the LSTM output and solved by the CRF layer. If you have already combined the CRF layer in your model but there are still some unreasonable outputs, you may need to train your model longer. Hope this helps.

**Comment by betterbear** [comment]
Hi, createmomo, the unreasonable outputs happens occasionally, not often. I think I have trained enough time, since the F1 score has raised to 95, and the network starts to overfit(train loss decreasing with validation loss rising slowly).

**Comment by createmomo** [comment]
Hi @betterbear , I am sorry for my late reply. Actually, I also had the same problem that the unreasonable outputs happened occasionally. One way to solve the problem I tried before was checking the training data to see whether there were some unreasonable labels in the training dataset. Sometimes, the train set was noisy. If the training data is all right but the problem still happens, I added a post-processing module in my project. Hope this helps!

**Comment by betterbear** [comment]
Hi, createmomo. Thanks very much. I used post-processing, too.  In my case, i can guarantee there is no unreasonable sequence in the training data. Because the BIO tag was generated by code ,and the original data was labeled in human readable format. Yesterday, I've read some paper which also suggested post-processing during decoding.

**Comment by createmomo** [comment]
Hi betterbear, thank you very much for the nice discussion. I believe your reply is also very beneficial to many other people who have the similar problem. :)

**Comment by susantiyuni** [comment]
Can I ask, what kind of post-processing here you talking about? Did you mean manually filtering the unreasonable sequence?
Thanks.
